### PR TITLE
feat: let clients toggle their own attachments

### PR DESCRIPTION
**Pull request by AntoineAugusti**
When looking to use attachments with my service for the first time I want to be able to toggle between different attachment options (including off) so that I can configure my Notify service in the way I want to use it.  

Trello: https://trello.com/c/y8pvFsDr/442-let-clients-toggle-their-own-attachments
Figma: https://www.figma.com/file/VuLnhytt6AJU92mQCNXbpK/File-attachments-%7C-Fichiers-joints?node-id=694%3A158

### CODE REVIEW

File: app/main/views/service_settings.py [hunk 1]
```diff
@@ -44,7 +44,6 @@
     SearchByNameForm,
```
Comment (AntoineAugusti) [inline app/main/views/service_settings.py:44]: Added a new endpoint to toggle on/off sending files and removed old code related to platform admin only + contact links (those are not required anymore)

File: app/main/views/service_settings.py [hunk 2]
```diff
@@ -73,7 +73,6 @@
 PLATFORM_ADMIN_SERVICE_PERMISSIONS = OrderedDict([
     ('inbound_sms', {'title': _l('Receive inbound SMS'), 'requires': 'sms', 'endpoint': '.service_set_inbound_number'}),
     ('email_auth', {'title': _l('Email authentication')}),
-    ('upload_document', {'title': _l('Send files by email'), 'endpoint': '.service_switch_can_upload_document'}),
```
Comment (AntoineAugusti) [inline app/main/views/service_settings.py:76]: Remove this line here, it was used to display this setting to platform admins

File: app/utils.py [hunk 3]
```diff
@@ -672,7 +672,7 @@ def yyyy_mm_to_datetime(string):
     return datetime(int(string[0:4]), int(string[5:7]), 1)
 
 
-def documentation_url(feature=None):
+def documentation_url(feature=None, section=None):
```
Comment (AntoineAugusti) [inline app/utils.py:675]: Added the possibility to link to a specific section within a page

File: tests/app/main/views/test_service_settings.py [hunk 4]
```diff
@@ -70,6 +70,7 @@ def mock_get_service_settings_page_common(
         'Send emails On Change',
```
Comment (AntoineAugusti) [inline tests/app/main/views/test_service_settings.py:70]: Added tests to show that there is a new line + a test for the controller. Kept old tests around but skipped those.

File: app/main/views/service_settings.py [hunk 5]
```diff
@@ -359,6 +358,32 @@ def service_switch_live(service_id):
...
+@user_has_permissions('manage_service')
+def service_switch_upload_document(service_id):
+    title = _("Send files by email")
+    form = ServiceOnOffSettingForm(
+        name=title,
+        enabled=current_service.has_permission('upload_document')
+    )
+    help = _(
+        "This feature is only available when sending through the API.<br>"
+        "Learn more in <a href='{}'>the API documentation</a>."
```
Comment (anikbrazeau) [inline app/main/views/service_settings.py:371]:
```suggestion
        "Learn more in the <a href='{}'>API documentation</a>."
```

File: app/main/views/service_settings.py [hunk 6]
```diff
@@ -359,6 +358,32 @@ def service_switch_live(service_id):
...
+    ).format(documentation_url('send', section='sending-a-file-by-email'))
+
+    if form.validate_on_submit():
+        current_service.force_permission('upload_document', on=form.enabled.data)
+        flash(_('Setting updated'), 'default')
```
Comment (anikbrazeau) [inline app/main/views/service_settings.py:376]: Is this something we do for any other setting?

File: app/translations/csv/fr.csv [hunk 7]
```diff
@@ -1317,3 +1317,6 @@
 "Once you've set up your team and tried out sending, you can request to go live in settings","Une fois que vous aurez mis en place votre équipe et essayé d'envoyer des messages, vous pourrez demander d'activer votre service dans les paramètres"
 "failure","échec"
 "failures","échecs"
+"This feature is only available when sending through the API.<br>Learn more in <a href='{}'>the API documentation</a>.","Cette fonctionnalité n’est disponible que pour les envois programmés par le biais de l’API.<br>Pour en savoir plus, <a href='{}'>consultez la documentation API</a>."
```
Comment (anikbrazeau) [inline app/translations/csv/fr.csv:1320]:
```suggestion
"This feature is only available when sending through the API.<br>Learn more in the <a href='{}'>API documentation</a>.","Cette fonctionnalité n’est disponible que pour les envois programmés par le biais de l’API.<br>Pour en savoir plus, consultez la <a href='{}'>documentation API</a>."
```

**Review by anikbrazeau — APPROVED** [review]
Looks good on heroku

**Comment by anikbrazeau** [comment]
Made change in docs here: https://github.com/cds-snc/notification-documentation/pull/74

**Comment by AntoineAugusti** [comment]
@QuanNguyen-CDS Would you be able to review this PR for the code side? Thanks 🙏

**Review by QuanNguyen-CDS — APPROVED** [review]
LGTM

**Pull request merged**